introspection/ws_server.go

```diff
@@ -0,0 +1,84 @@
+package introspection
+
+import (
+	"context"
+	"github.com/gogo/protobuf/proto"
+	"github.com/gorilla/websocket"
+	logging "github.com/ipfs/go-log"
+	"github.com/libp2p/go-libp2p-core/introspect"
+	"net/http"
+	"time"
+)
+
+var logger = logging.Logger("introspection-server")
+var upgrader = websocket.Upgrader{}
+
+// StartServer starts the ws introspection server with the given introspector
+func StartServer(introspector introspect.Introspector) func() error {
+	// introspect handler
+	http.HandleFunc("/introspect", toHttpHandler(introspector))
+
+	// start server
+	serverInstance := http.Server{
+		// TODO Need a better strategy to select an address
```

Comment:
Let's make sure this doesn't fall off the radar by opening an issue.

Reply:
Done in https://github.com/libp2p/go-libp2p-introspection/issues/6